signature/ecc-sign-verify/ecc_sign_verify.c

```diff
@@ -0,0 +1,166 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <stdio.h>
+#include <wolfssl/options.h>
+
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/signature.h>
+#include <wolfssl/wolfcrypt/hash.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+/* uncomment to show signatures */
+/* #define SHOW_SIGS_IN_EXAMPLE */
+
+#define HEAP_HINT NULL
+#define ECC_KEY_SIZE_112 112
+#define ECC_KEY_SIZE_128 128
+#define ECC_KEY_SIZE_160 160
+#define ECC_KEY_SIZE_192 192
+#define ECC_KEY_SIZE_224 224
+#define ECC_KEY_SIZE_239 239
+#define ECC_KEY_SIZE_256 256
+#define ECC_KEY_SIZE_320 320
+#define ECC_KEY_SIZE_384 384
+#define ECC_KEY_SIZE_512 512
+#define ECC_KEY_SIZE_521 521
+#define BYTE_SZ 8
+
+#define CHECK_RET(a, b, eLabel, msg) { \
+                                        if (a != b) {                    \
+                                            printf("failed %s\n", msg);  \
+                                            goto eLabel;                 \
+                                        }                                \
+                                     }
+
+int do_sig_ver_test(int eccKeySz);
+
+#ifdef SHOW_SIGS_IN_EXAMPLE
+    static void hexdump(const void *buffer, word32 len, byte cols);
+#endif
+
+int main(void)
+{
+    int ret = 0;
+    ret = do_sig_ver_test(ECC_KEY_SIZE_112);
+    CHECK_RET(ret, 0, finished, "112 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_128);
+    CHECK_RET(ret, 0, finished, "128 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_160);
+    CHECK_RET(ret, 0, finished, "160 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_192);
+    CHECK_RET(ret, 0, finished, "192 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_224);
+    CHECK_RET(ret, 0, finished, "224 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_239);
+    CHECK_RET(ret, 0, finished, "239 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_256);
+    CHECK_RET(ret, 0, finished, "256 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_320);
+    CHECK_RET(ret, 0, finished, "320 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_384);
+    CHECK_RET(ret, 0, finished, "384 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_512);
+    CHECK_RET(ret, 0, finished, "512 test");
+    ret = do_sig_ver_test(ECC_KEY_SIZE_521);
+    CHECK_RET(ret, 0, finished, "521 test");
+finished:
+    return ret;
+}
+
+int do_sig_ver_test(int eccKeySz)
+{
+    int ret;
+    unsigned char hash[32] = "A 32-bit string to test signing";
+    ecc_key key;
+    byte* sig = NULL; // get rid of this magic number
+    WC_RNG rng;
+    int verified = 0;
+    /* for odd curve sizes account for mod EG (256/8) + (256%8) == 32 + 0 = 32
+     * alternatively (521/8 = 65.125 (rounds to 65) + (521%8) == 65 + 1 = 66
+     */
+    int byteField = (eccKeySz / BYTE_SZ) + (((eccKeySz % BYTE_SZ) != 0) ? 1:0);
```

Comment:
This could also be written as:
`byteField = ((eccKeySz + (BYTE_SZ-1)) / BYTE_SZ);`

Reply:
Implemented.